content/en/blog/2022/how-to-get-started-with-opentelemetry-go.md

```diff
@@ -0,0 +1,420 @@
+    "go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin"
+    "go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo"
+    "go.opentelemetry.io/otel"
+    "go.opentelemetry.io/otel/propagation"
+)
+var client * mongo.Client
+func main() { 
+    //Export traces to Jaeger
+    tp, tpErr: = tracing.JaegerTraceProvider()
+    if tpErr != nil {
+        log.Fatal(tpErr)
+    }
+    otel.SetTracerProvider(tp)
+    otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext {}, propagation.Baggage {}))
+    connectMongo()
+    startWebServer()
+}
+func connectMongo() {
+    opts: = options.Client()
+    //Mongo OpenTelemetry instrumentation
+    opts.Monitor = otelmongo.NewMonitor()
+    opts.ApplyURI("mongodb://localhost:27017")
+    client, _ = mongo.Connect(context.Background(), opts)
+    //Seed the database with some todo's
+    docs: = [] interface {} {
+        bson.D {
+                {
+                    "id", "1"
+                }, {
+                    "title", "Buy groceries"
+                }
+            },
+            bson.D {
+                {
+                    "id", "2"
+                }, {
+                    "title", "install Aspecto.io"
+                }
+            },
+            bson.D {
+                {
+                    "id", "3"
+                }, {
+                    "title", "Buy dogz.io domain"
+                }
+            },
+    }
+    client.Database("todo").Collection("todos").InsertMany(context.Background(), docs)
+}
+func startWebServer() {
+    r: = gin.Default()
+    //gin OpenTelemetry instrumentation
+    r.Use(otelgin.Middleware("todo-service"))
+    r.GET("/todo", func(c * gin.Context) {
+        collection: = client.Database("todo").Collection("todos")
+        //Make sure to pass c.Request.Context() as the context and not c itself
+        cur, findErr: = collection.Find(c.Request.Context(), bson.D {})
+        if findErr != nil {
+            c.AbortWithError(500, findErr)
+            return
+        }
+        results: = make([] interface {}, 0)
+        curErr: = cur.All(c, & results)
+        if curErr != nil {
+            c.AbortWithError(500, curErr)
+            return
+        }
+        c.JSON(http.StatusOK, results)
+    })
+    _ = r.Run(":8080")
+}
+```
+
+### Export traces to Jaeger
+
+1. Run the todo-service with go run main.go
+2. Make an HTTP GET request to localhost:8080/todo to generate some traces in Go
+3. Open Jaeger at http://localhost:16686/search to view those traces
+
+You can now see the Jaeger UI. Select todo-service and click on Find traces. You should see your trace on the right:
+!["Jaeger UI displays opentelemetry traces in go for our todo-service"](https://lh5.googleusercontent.com/ZeFbAE9-XVSc-5GHjZkslHuJ3f01VQqSrObOgLY9yDSjuTyJdvvAzIapIvTQqumFTUP2BZE4gxd-Vt2JXjvqO1ep3JUBhkHKiry_m8bSAwwvEf3kKNfzFiKwzFP8E3btWtQV0pLZZWnsbY-sUA)
```

Comment:
This and the next image are difficult to read. Could you regenerate some images but make the window that you're taking a screenshot of narrower?